Marlin/Marlin_main.cpp

```diff
@@ -8556,29 +8545,137 @@ inline void gcode_M907() {
 
 #if HAS_CASE_LIGHT
 
-  uint8_t case_light_brightness = 255;
+  static bool useable_hardware_PWM(uint8_t pin) {
+
+    #if !defined(TIMER1B)  // working with Teensyduino extension so need to define the timers
+      #define TIMER0B 1
+      #define TIMER1A 7
+      #define TIMER1B 8
+      #define TIMER1C 9
+      #define TIMER2A 6
+      #define TIMER2B 2
+      #define TIMER3A 5
+      #define TIMER3B 4
+      #define TIMER3C 3
+    #endif
+
+
+  /**
+   *      Timer   Digital Normally    Used by         Optional
+   *      & Port   Pin    assigned    system          users
+   *      TIMER3B   2     X_MAX
+   *      TIMER3C   3     X_MIN
+   *      TIMER0B   4     HEATER_4    #temp & milli ISR
+   *      TIMER3A   5     HEATER_5                    *servo 0-11 ISR
+   *      TIMER4A   6     HEATER_6                    *servo 12-23 ISR
+   *      TIMER4B   7     LCD
+   *      TIMER4C   8     HOTBED
+   *      TIMER2B   9     HEATER_1
+   *      TIMER2A   10    HEATER_0
+   *      TIMER1A   11    HEATER_7    *stepper ISR     *E axis waveform generator
+   *      TIMER1B   12    PS_ON_PIN   *stepper ISR
+   *      TIMER0A   13    LED         LED PWM & #step adv ISR
+   *      TIMER1C   13                *stepper ISR
+   *      TIMER5C   44    LCD                         stepper motor current XY PWM
+   *      TIMER5B   45    LCD                         stepper motor current Z PWM
+   *      TIMER5A   46    Z_STEP                      stepper motor current E PWM or *servo 24-35 ISR
+   *
+   *   * - These hardware PWMs are not available.  The pin can still be used as a general purpose
+   *       digital I/O.
+   *
+   *   # - still can be used a hardware PWM even though it's also used by an ISR
+   */
+
+
+    switch(digitalPinToTimer(pin)) {
```

Comment:
Add a space between `switch` and the opening parenthesis.
Only _functions_ should snuggle with the opening parenthesis.
http://marlinfw.org/docs/development/coding_standards.html#spacing